parquet-hadoop/src/main/java/org/apache/parquet/hadoop/api/ReadSupport.java

```diff
@@ -75,14 +76,32 @@ public ReadContext init(
     throw new UnsupportedOperationException("Override init(InitContext)");
   }
 
+  /**
+   * called in {@link org.apache.hadoop.mapreduce.InputFormat#getSplits(org.apache.hadoop.mapreduce.JobContext)} in the front end
+   *
+   * @param configuration    the configuration
+   * @param keyValueMetaData the app specific metadata from the file
+   * @param fileSchema       the schema of the file
+   * @return the readContext that defines how to read the file
+   *
+   * @deprecated override {@link ReadSupport#init(InitContext)} instead
+   */
+  @Deprecated
```

Comment:
Do we still need to add a `deprecated` method?

Reply:
This PR is focussed on transitioning from `Configuration` to the `ParquetConfiguration` interface. This included some calls to deprecated methods which I could not very quickly transition away from. I would consider this out-of-scope for this PR.